source_handlers_test.go

```diff
@@ -275,6 +276,85 @@ func TestSourceListSatellite(t *testing.T) {
 	AssertLinks(t, c.Request().RequestURI, out.Links, 100, 0)
 }
 
+func TestSourceListInternal(t *testing.T) {
+	if !flags.Integration {
+		t.Skip("Only run during integration tests")
+	}
+
+	c, rec := request.CreateTestContext(
+		http.MethodGet,
+		"/internal/v2.0/sources",
+		nil,
+		map[string]interface{}{
+			"limit":   100,
+			"offset":  0,
+			"filters": []util.Filter{},
+		})
+
+	err := InternalSourceList(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != 200 {
+		t.Error("Did not return 200")
+	}
+
+	var out util.Collection
+	err = json.Unmarshal(rec.Body.Bytes(), &out)
+	if err != nil {
+		t.Error("Failed unmarshaling output")
+	}
+
+	if out.Meta.Limit != 100 {
+		t.Error("limit not set correctly")
+	}
+
+	if out.Meta.Offset != 0 {
+		t.Error("offset not set correctly")
+	}
+
+	if len(out.Data) != len(fixtures.TestSourceData) {
+		t.Error("not enough objects passed back from DB")
+	}
+
+	for i, src := range out.Data {
```

Comment:
maybe do we need to sort those `out.Data`  for sure ? function `SortByStringValueOnKey("id", out.Data)` could be used for it. - I don't whether that is defined order for `out.Data`

Reply:
I'm sorry: I've tried to find the function you told me about, but I am not able to find it :eyes: 

Anyway: in this case Gorm seems to be respecting the order we set in the fixtures, so we shouldn't have any problem with the tests. Although we could create a `TODO` for this to be safer. What do you think?